### PR TITLE
Make Figure a subtype of GtkWidget and reduce icon size

### PR DESCRIPTION
**Pull request by jonathanBieler**
This allow to easily integrate Figures into other GtkContainers, cf #58

### CODE REVIEW

File: src/display_gadfly.jl [hunk 1]
```diff
@@ -33,20 +33,30 @@ initialized(pzc::PanZoomCallbacks) = pzc.idpzk != 0
...
+    tb::Gtk.GtkToolbar
     prepped         # tuple, a pre-processed Plot to speed rendering
     cc::Compose.Context   # fully-rendered Plot (useful for hit-testing)
     tweaks::Dict{Tuple{Symbol,Symbol},Any}  # post-rendering property changes
     panzoom_cb::PanZoomCallbacks
     figno::Int
 
-    function Figure(c::GtkCanvas, p::Plot)
+    function Figure(p::Plot)
+
```
Comment (timholy) [inline src/display_gadfly.jl]: Dumb point, but I'd prefer not to have a blank line at the beginning of functions

File: src/display_gadfly.jl [hunk 2]
```diff
@@ -287,23 +301,29 @@ closefig(i::Integer) = (fig = getfig(_display,i); clear_hit(fig); gtkdestroy(get
...
-    Gtk.GAccessor.icon_widget(lasso_button, Gtk.@GtkImage(joinpath(HOME, "images", "lasso_icon.png")))
+    Gtk.GAccessor.icon_widget(lasso_button, Gtk.@GtkImage(joinpath(Immerse.HOME, "images", "lasso_icon_16.png")))
     push!(tb, save_as)
     push!(tb, Gtk.@GtkSeparatorToolItem())
     push!(tb, zb)
     push!(tb, fullview)
     push!(tb, Gtk.@GtkSeparatorToolItem())
     push!(tb, lasso_button)
+
+    Gtk.G_.icon_size(tb,1)#small icon size
```
Comment (timholy) [inline src/display_gadfly.jl]: Thank you for solving this problem!

**Comment by timholy** [comment]
LGTM. If you can fix the (dumb) spacing issue and ideally squash out the noise (if you're git-comfortable; don't worry about it if not), then let's merge this.

**Comment by jonathanBieler** [comment]
Done, I had some troubles with the squashing, but I think I did it right in the end.

**Comment by timholy** [comment]
Many thanks! I'm excited to make use of this.

**Comment by jonathanBieler** [comment]
Great, it makes for a pretty good plotting solution out of the box.

**Pull request merged**